src/app/projects/components/project-settings/components/labeling-tasks/labeling-tasks.component.ts

```diff
@@ -0,0 +1,304 @@
+      func: this.handleWebsocketNotification
+    });
+  }
+
+  ngAfterViewInit() {
+    this.inputTaskName.changes.subscribe(() => {
+      this.setFocus(this.inputTaskName);
+    });
+  }
+
+  private setFocus(focusArea) {
+    if (focusArea.length > 0) {
+      focusArea.first.nativeElement.focus();
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions$.forEach((subscription) => subscription.unsubscribe());
+    NotificationService.unsubscribeFromNotification(this, this.project.id);
+  }
+
+  prepareLabelingTasksRequest(projectId: string) {
+    let labelingTasks$;
+    [this.labelingTasksQuery$, labelingTasks$] = this.projectApolloService.getLabelingTasksByProjectId(projectId);
+    this.subscriptions$.push(labelingTasks$.subscribe((tasks) => {
+      tasks.sort((a, b) => a.relativePosition - b.relativePosition || a.name.localeCompare(b.name))
+
+      if (this.onlyLabelsChanged(tasks)) {
+        this.lh.setLabelMap(tasks);
+      } else {
+        this.labelingTasksArray.clear();
+        tasks.forEach((task) => {
+          task.labels.sort((a, b) => a.name.localeCompare(b.name));
+          this.lh.labelingTaskColors.set(task.id, task.labels.map((label) => label.color));
+          task.labels = task.labels.map((label) => this.lh.extendLabelForColor({ ...label }));
+          let group = this.formBuilder.group({
+            id: task.id,
+            name: task.name,
+            nameOpen: false,
+            targetId: task.taskTarget == LabelingTaskTarget.ON_ATTRIBUTE ? task.attribute.id : "",
+            targetName: task.taskTarget == LabelingTaskTarget.ON_ATTRIBUTE ? task.attribute.name : "Full Record",
+            taskType: task.taskType,
+          })
+          this.lh.labelMap.set(task.id, task.labels);
+          group.valueChanges.pipe(distinctUntilChanged()).subscribe(() => {
+            let values = group.getRawValue(); //to ensure disabled will be returned as well
+            if (values.nameOpen || !this.isTaskNameUniqueCheck(values.name, group) || values.name.trim() == "") return;
+            this.projectApolloService.
+              updateLabelingTask(this.project.id, values.id, values.name, values.taskType, values.targetId == "" ? null : values.targetId).pipe(first()).subscribe();
+          });
+          this.labelingTasksArray.push(group);
+        });
+      }
+
+    }));
+    return labelingTasks$;
+  }
+
+  addLabelingTaskModal() {
+    this.settingModals.labelingTask.create.open = true;
+    this.dataHandlerHelper.focusModalInputBox('labelingTaskName');
+  }
+
+  labelingTasksDropdownValues() {
+    if (this.labelingTasksDropdownArray.length == 0) {
+      for (let t of Object.values(LabelingTask)) {
+        if (t == LabelingTask.NOT_USEABLE) continue;
+        this.labelingTasksDropdownArray.push({
+          name: labelingTaskToString(t),
+          value: t,
+        });
+      }
+    }
+    return this.labelingTasksDropdownArray;
```

Comment:
this is old logic used to always have an initialized array with the method (common c# style) since we don't want to use function in the HTML anymore it would be better to initialize the labelingTasksDropdownArray in the oninit use it in the HTML and be done with it